spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapsCache.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.client.config;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.kubernetes.commons.config.ConfigMapCache;
+import org.springframework.cloud.kubernetes.commons.config.StrippedSourceContainer;
+import org.springframework.core.log.LogAccessor;
+
+/**
+ * A cache of V1ConfigMap(s) per namespace. Makes sure we read config maps only once from
+ * a namespace.
+ *
+ * @author wind57
+ */
+final class KubernetesClientConfigMapsCache implements ConfigMapCache {
```

Comment:
what we used to do is : 

- read configmaps
- strip them down to the properties that we care for

what we do now:

- read configmaps
- strip them down to the properties that we care for
- cache those stripped sources

So instead of simply caching `List<V1ConfigMap>`, we now cache `List<StrippedSourceContainer>`, which is basically `strippedConfigMaps(configMaps)`. it's the same logical thing

Reply:
This process is the same for both clients, for bot configmaps and secrets